### PR TITLE
add .summary.social and laborObligations codelist

### LINKED ISSUE #404: Labor obligations: record labor obligations included in contracts in new social object
# Background

This issue relates to the following CoST IDS elements proposed in the CoST IDS/OC4IDS review:

<details>

<summary> Labour obligations </summary>

## Labour obligations

**Module:** Social
**Indicator:** Labour compliance

- Minimum wage 
- Overtime 
- Prohibition of forced labour 
- Prohibition of child labour 
- Equal opportunity 
- Non-discrimination 
- Freedom of association 
- Grievance mechanism 
- Working at height 
- Underground work 
- Handling of materials/equipment 
- Monitoring of accidents 
- Traffic management 
- Accommodation 
- Protective equipment 
- Others (explain) 

> Contracting process level: Publish a summary of the labor obligations: 
1. For each labor obligation in the contract, add a code from the laborObligations codelist to the`.summary.social.laborObligations.obligations` array. 
2. Optionally, add a further explanation of the labor obligations to `.summary.social.laborObligations.description`. Publish the bidding documents that specify labor obligations: Add a document to `.summary.documents`, set its `.id` incrementally, set its `.documentType` to ‘biddingDocuments’ and set its `.url` to the URL at which the documents are available. Publish the signed contract that includes labor obligations:  Add a document to `.summary.documents`, set its `.id` incrementally, set its `.documentType` to ‘contractSigned’ and set its `.url` to the URL at which the signed contract that includes labor obligations is accessible. 

</details>

# Proposal

Create an `laborObligations` object within the new `social` object (see overarching [issue](https://github.com/open-contracting/infrastructure/issues/403)). This shall include an `obligations` array of codes taken from the open ‘laborObligations’ codelist, and a `description` field. 

Research is still required to fully define the codelist

Add the following fields and objects:

<div dir="ltr" style="margin-left:0pt;" align="left" id="docs-internal-guid-6c7fc696-7fff-bbe9-c51b-a9feb3a2b98b">

Path | Title | Description | Type (format) | Codelist
-- | -- | -- | -- | --
social.laborObligations | Labor obligations | Information about labor obligations stipulated in the construction contract(s). | object | -
laborObligations.obligations | Obligations | The labor obligations stipulated in the construction contract(s), using the open laborObligations codelist. | array of strings | laborObligations
laborObligations.description | Description | A short description of the labor obligations | string | -

</div>

Add the following codes:

<div dir="ltr" style="margin-left:0pt;" align="left" id="docs-internal-guid-b8bcb924-7fff-8b4a-55b7-2cc2c640397a">

Codelist | Code | Title | Description
-- | -- | -- | --
laborObligations | minimumWage | Minimum wage | TBD
laborObligations | overtime | Overtime | TBD
laborObligations | prohibitionForcedLabor | Prohibition of forced labor | TBD
laborObligations | prohibitionChildLabor | Prohibition of child labor | TBD
laborObligations | equalOpportunities | Equal opportunities | TBD
laborObligations | nondiscrimination | Non-discrimination | TBD
laborObligations | grievanceMechanism | Grievance mechanism | TBD
laborObligations | workingAtHeight | Working at height | TBD
laborObligations | undergroundWork | Underground work | TBD
laborObligations | handling | Handling of materials/equipment | TBD
laborObligations | accidentMonitoring | Monitoring of accidents | TBD
laborObligations | trafficManagement | Traffic management | TBD
laborObligations | accommodation | Accommodation | TBD
laborObligations | protectivePersonalEquipment | Protective personal equipment | TBD

</div>

# Example

```json

{ 
 "contractingProcesses": [ 
   { 
 	"id": "1", 
 	"summary": { 
   	"social": { 
     	"laborObligations": { 
       	"obligations": [ 
         	"minimumWage", 
         	"paidOvertime" 
       	], 
       	"description": "The contract's labor obligations include a minimum wage of $20 per hour and an overtime limit of 10 hours per week." 
     	} 
   	} 
     } 
   }
 ]
}

```

# Sources
* [SuRE - Standard for Sustainable and Resilient Infrastructure](https://sure-standard.org/wp-content/uploads/2021/07/ST01_Normative_Standard_v2.0_clean.pdf)
* [World Bank Environmental and Social framework](https://thedocs.worldbank.org/en/doc/837721522762050108-0290022018/original/ESFFramework.pdf)
* [Equator principles](https://equator-principles.com/app/uploads/The-Equator-Principles_EP4_July2020.pdf) - specifically for Grievance mechanisms
* [GRESB 2023 Infrastructure Asset Standard and Reference Guide](https://documents.gresb.com/generated_files/infrastructure/2023/asset/reference_guide/complete.html#performance-health-and-safety)

cc @evelyndinora

### LINKED ISSUE #406: labor budget: specify the amount set aside in the contract to cover labor costs
# Background

This issue relates to the following CoST IDS elements proposed in the CoST IDS/OC4IDS review:

<details>

<summary> Labor budget </summary>

## Labor budget

**Module:** Social
**Indicator:** Labor compliance

### Disclosure format

> Disclose the amount allocated by the main contractor to cover for labour costs (E.g. [value]). 

### OC4IDS mapping

> Contract level: Publish the amount and currency of the labor budget in `.summary.social.laborBudget`. 

</details>

# Proposal

Add a `Value` object, `laborBudget` to the new `social` object (see #403).

Add the following fields and objects:

<div dir="ltr" style="margin-left:0pt;" align="left" id="docs-internal-guid-450302af-7fff-3230-8cd4-ef9d69553192">

Path | Title | Description | Type (format) | Codelist
-- | -- | -- | -- | --
social.laborBudget | Labor budget | The budget allocated in the contract for labor costs. | `Value` object | -

</div>

# Example

```json

{ 
 "contractingProcesses": [ 
   { 
 	"id": "1", 
 	"summary": { 
   	"social": { 
     	"laborBudget": { 
       	"amount": 10000000, 
       	"currency": "USD" 
     	} 
   	} 
 	} 
   } 
 ] 
} 

```

## Discussion
* Assuming that the use case for this data point is to do with making sure that labour costs are appropriate, it’s difficult to see what the value of this data point is without also knowing at least the number of person-hours the amount is supposed to cover. This need can be met through comparison or procurement and contract documents so this data point possibly isn’t needed.
* Answer - Over time this data point will allow a comparison of the contracts where there is no labour budget specified vs. contracts where a labour budget exits. This will be very useful information to be assessed in combination with the following data point (health and safety – workers’ accidents). I see a value here from a policy perspective. From interviews some contractors do specify this in their contracts, labor in general covering e.g. PPE costs, overtime, salaries, accomodation. Not a common thing in the contracts. Want to emphasis where there is money protected for labor issues, not really about checking for if a contractor is paying their laborers well, or comparing across projects.

cc @evelyndinora

### PR DESCRIPTION
**Pull request by odscjen**
**Related issues**
closes #404 
closes #406 

**Description**

altered the descriptions in the laborObligations codelist to use non-normative language as suggested in the issue.

**Merge checklist**

- [x] [Log your changes](https://ocds-standard-development-handbook.readthedocs.io/en/latest/standard/contributing.html#logging-changes)

If there are changes to `project-schema.json` or `project-package-schema.json`:

- [x] Update the examples:
  - [ ] `docs/examples/example.json`
  - [ ] `docs/examples/blank.json`
- [x] Run `./manage.py pre-commit` to update `docs/_static/i8n.csv`

If you added a new definition to the schema, run `./manage.py pre-commit`.

If you added a new codelist:

  - [x] Add an entry to `docs/reference/codelists.md`

### CODE REVIEW

File: schema/project-level/codelists/laborObligations.csv [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Code,Title,Description,Source
+minimumWage,Minimum wage,"A minimum wage that an employer needs to pay their labor force for work performed, which cannot be reduced by collective or individual agreements or contracts.",OC4IDS
+paidOvertime,Paid overtime,Agreement that hours worked in addition to regular working hours need to be paid at an specified level.,OC4IDS
+weeklyRestProtection,Weekly rest protection,Regular working hours and voluntary overtime are both capped to protect a minimum level of weekly rest.,OC4IDS
+forcedLaborProhibition,Prohibition of forced labor,The project needs to comply with all applicable national and international laws regarding forced or compulsory labor and not resort to any form of forced labor.,OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
forcedLaborProhibition,Prohibition of forced labor,The project will not resort to any form of forced labor and will comply with all applicable national and international laws regarding forced or compulsory labor.,OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+childLaborProhibition,Prohibition of child labor,The project needs to comply with all applicable national and international laws regarding employment of minors and not resort to any form of child labor.,OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
childLaborProhibition,The project will not resort to any form of child labor and will comply with all applicable national and international laws regarding employment of minors.,OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 3]
```diff
@@ -0,0 +1,16 @@
+Code,Title,Description,Source
+minimumWage,Minimum wage,"A minimum wage that an employer needs to pay their labor force for work performed, which cannot be reduced by collective or individual agreements or contracts.",OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
minimumWage,Minimum wage,"A minimum wage that an employer will pay their labor force for work performed, which cannot be reduced by collective or individual agreements or contracts.",OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+paidOvertime,Paid overtime,Agreement that hours worked in addition to regular working hours need to be paid at an specified level.,OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
paidOvertime,Paid overtime,Agreement that hours worked in addition to regular working hours will be paid at a specified level.,OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+weeklyRestProtection,Weekly rest protection,Regular working hours and voluntary overtime are both capped to protect a minimum level of weekly rest.,OC4IDS
+forcedLaborProhibition,Prohibition of forced labor,The project needs to comply with all applicable national and international laws regarding forced or compulsory labor and not resort to any form of forced labor.,OC4IDS
+childLaborProhibition,Prohibition of child labor,The project needs to comply with all applicable national and international laws regarding employment of minors and not resort to any form of child labor.,OC4IDS
+equalOpportunities,Equal opportunities,Employment opportunities need to not be based on personal characteristics unrelated to job requirements.,OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
equalOpportunities,Equal opportunities,Employment opportunities will not be based on personal characteristics unrelated to job requirements.,OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+nondiscrimination,Non-discrimination,"The project needs to not discriminate against any employees with respect to any aspects of employment, such as hiring, wages, access to training, job assignment, promotion, or termination of employment.",OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
nondiscrimination,Non-discrimination,"The project will not discriminate against any employees with respect to any aspects of employment, such as hiring, wages, access to training, job assignment, promotion, or termination of employment.",OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+grievanceMechanism,Grievance mechanism,The project needs to provide accessible and transparent grievance mechanisms to enable all workers to raise workplace concernts.,OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
grievanceMechanism,Grievance mechanism,The projectwill provide accessible and transparent grievance mechanisms to enable all workers to raise workplace concernts.,OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+workingAtHeight,Working at height,Adequate fall prevention and protection measures needs to be implemented wherever workers are expected to work at height.,OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
workingAtHeight,Working at height,Adequate fall prevention and protection measures will be implemented wherever workers are expected to work at height.,OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+undergroundWork,Underground work,"The project needs to provide adequate occupational health and safety equipment, information and services appropriate for underground and excavation work such as drilling, blasting, rock bolting and soft soil work.",OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
undergroundWork,Underground work,"The project will provide adequate occupational health and safety equipment, information and services appropriate for underground and excavation work such as drilling, blasting, rock bolting and soft soil work.",OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+materialsEquipmentHandling,Handling of materials/equipment,The project needs to ensure that all workers engaged in handling of materials or equipment as part of their employment will receive at no cost to themselves adequate training and appropriate protective equipment.,OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
materialsEquipmentHandling,Handling of materials/equipment,The project will ensure that all workers engaged in handling of materials or equipment as part of their employment will receive at no cost to themselves adequate training and appropriate protective equipment.,OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+accidentMonitoring,Monitoring of accidents,"The project needs to monitor, document and report on all occupational accidents, diseases and incidents.",OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
accidentMonitoring,Monitoring of accidents,"The project will monitor, document and report on all occupational accidents, diseases and incidents.",OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+trafficManagement,Traffic management,The project needs to adopt road safety initiatives proportional to its scope and nature that are protective of project workers and other road users.,OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
trafficManagement,Traffic management,The project will adopt road safety initiatives proportional to its scope and nature that are protective of project workers and other road users.,OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+accommodation,Accommodation,"The project needs to provide accommodation services to workers that accommodate their physical, social and cultural needs and protect and promote their health, safety and well-being.",OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
accommodation,Accommodation,"The project will provide accommodation services to workers that accommodate their physical, social and cultural needs and protect and promote their health, safety and well-being.",OC4IDS
```

File: schema/project-level/codelists/laborObligations.csv [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+protectivePersonalEquipment,Protective personal equipment,"The project needs to provide access to, training in, and maintenance of appropriate personal protective equipment at no additional cost to its workers.",OC4IDS
```
Comment (duncandewhurst) [inline schema/project-level/codelists/laborObligations.csv]:
```suggestion
protectivePersonalEquipment,Protective personal equipment,"The project will provide access to, training in, and maintenance of appropriate personal protective equipment at no additional cost to its workers.",OC4IDS
```

File: mapping/sustainability.yaml [hunk 15]
```diff
@@ -1335,7 +1335,7 @@
     module: Social
     indicator: Labour compliance
     disclosure format: Disclose the amount allocated by the main contractor to cover for labour costs (E.g. \[value\]).
-    mapping: 'Contract level: Publish the amount and currency of the labor budget in `.summary.social.laborBudget`.'
+    mapping: 'Contracting process level: Publish the amount and currency of the labor budget in `.summary.social.laborBudget`.'
```
Comment (duncandewhurst) [inline mapping/sustainability.yaml]:
```suggestion
    mapping: 'Contracting process level: Map to `.summary.social.laborBudget`.'
```

**Review by duncandewhurst — CHANGES_REQUESTED** [review]
'will' isn't a normative keyword so I suggested reverting most of the changes to the code descriptions to reinstate it because the 'needs to' alternatives didn't read as well. If I understood it correctly, https://github.com/open-contracting/infrastructure/issues/404#issuecomment-1808505058, was only about changing instances of 'must', 'shall' etc. and pointing out that most code descriptions already used 'will'.

Please update the mapping for labor obligations to add a link to the codelist and to reference the common operation for adding a contracting process document.

**Comment by odscjen** [comment]
ah, I'd read it as saying that "will" should be removed, I've reverted them all back as suggested.

File: mapping/sustainability.yaml [hunk 16]
```diff
@@ -1293,12 +1293,12 @@
 
         Publish a summary of the labor obligations:
 
-        1. For each labor obligation in the contract, add a code from the laborObligations codelist to the`.summary.social.laborObligations.obligations` array.
+        1. For each labor obligation in the contract, add a code from the [laborObligations](https://standard.open-contracting.org/infrastructure/%7B%7Bversion%7D%7D/%7B%7Blang%7D%7D/reference/codelists/#laborobligations) codelist to the`.summary.social.laborObligations.obligations` array.
```
Comment (duncandewhurst) [inline mapping/sustainability.yaml]: The mappings are rendered as Markdown on `sustainability.md` so we should use relative links, same as on other Markdown pages.

```suggestion
        1. For each labor obligation in the contract, add a code from the [laborObligations](../../reference/codelists.md#laborobligations) codelist to the`.summary.social.laborObligations.obligations` array.
```

File: mapping/sustainability.yaml [hunk 17]
```diff
@@ -1293,12 +1293,12 @@
...
         2. Optionally, add a further explanation of the labor obligations to `.summary.social.laborObligations.description`.
 
-        Publish the bidding documents that specify labor obligations: Add a document to `.summary.documents`, set its `.id` incrementally, set its `.documentType` to 'biddingDocuments' and set its `.url` to the URL at which the documents are available.
+        Publish the bidding documents that specify labor obligations: [Add a contracting process document](../common.md#add-a-contracting-process-document), to `.summary.documents`, set its `.id` incrementally, set its `.documentType` to 'biddingDocuments' and set its `.url` to the URL at which the documents are available.
```
Comment (duncandewhurst) [inline mapping/sustainability.yaml]: Remove content covered by common operation.

```suggestion
        Publish the bidding documents that specify labor obligations: [Add a contracting process document](../common.md#add-a-contracting-process-document) and set its `.documentType` to 'biddingDocuments'.
```

File: mapping/sustainability.yaml [hunk 18]
```diff
@@ -1293,12 +1293,12 @@
...
 
-        Publish the signed contract that includes labor obligations:  Add a document to `.summary.documents`, set its `.id` incrementally, set its `.documentType` to 'contractSigned' and set its `.url` to the URL at which the signed contract that includes labor obligations is accessible.
+        Publish the signed contract that includes labor obligations:  [Add a contracting process document](../common.md#add-a-contracting-process-document), to `.summary.documents`, set its `.id` incrementally, set its `.documentType` to 'contractSigned' and set its `.url` to the URL at which the signed contract that includes labor obligations is accessible.
```
Comment (duncandewhurst) [inline mapping/sustainability.yaml]: Remove content covered by common operation.

```suggestion
        Publish the signed contract that includes labor obligations:  [Add a contracting process document](../common.md#add-a-contracting-process-document) and set its `.documentType` to 'contractSigned'.
```

**Review by duncandewhurst — CHANGES_REQUESTED** [review]
Happy for you to request a review from James once my comments are resolved.

**Comment by jpmckinney** [comment]
Please re-request review from me after Duncan's review is resolved.

File: mapping/sustainability.yaml (hunk 16 above)
Comment (odscjen) [reply mapping/sustainability.yaml]: ah, the one example I could find in the reviewed PR's at this point as them as full urls, so you'll find a few of these in various PR's I've done up to now

**Review by jpmckinney — APPROVED** [review]

**Pull request merged**